packages/video_player/video_player_avfoundation/ios/Classes/FLTVideoPlayerPlugin.m

```diff
@@ -317,6 +322,11 @@ - (void)observeValueForKeyPath:(NSString *)path
     if (_eventSink != nil) {
       _eventSink(@{@"event" : @"bufferingEnd"});
     }
+  } else if (context == rateContext) {
+    AVPlayer *player = (AVPlayer *)object;
```

Comment:
maybe add a comment that this is different from other KVO, since it's the only one not observing the item. It feels very easy to mess it up by accident in the future.